operator/src/main/java/org/keycloak/operator/Utils.java

```diff
@@ -77,4 +80,35 @@ public static <T extends HasMetadata> Optional<T> getByName(Class<T> clazz, Func
         return ies.get(new ResourceID(nameFunction.apply(primary), primary.getMetadata().getNamespace()));
     }
 
+    /**
+     * Set resources requests/limits for Keycloak container
+     * </p>
+     * If not specified in the Keycloak CR, set default values from operator config
+     */
+    public static void addResources(ResourceRequirements resource, Config config, Container kcContainer) {
+        final ResourceRequirements resourcesSpec = Optional.ofNullable(resource).orElseGet(ResourceRequirements::new);
+
+        // sets the min boundary when the spec is not present
+        final var requests = Optional.ofNullable(resourcesSpec.getRequests()).orElseGet(HashMap::new);
+
+        final var requestsMemory = requests.get("memory");
+        final var defaultRequestsMemory = config.keycloak().resources().requests().memory();
+
+        // Validate 'requests' memory
+        if (requestsMemory != null) {
+            var specifiedMemoryIsLessThanDefault = requestsMemory.getNumericalAmount().intValue() < defaultRequestsMemory.getNumericalAmount().intValue();
+            if (specifiedMemoryIsLessThanDefault) {
+                Log.warnf("Provided 'requests' memory ('%s') is less than used default value ('%s'). Use it in your risk, as Keycloak performance might be degraded.", requestsMemory, defaultRequestsMemory);
```

Comment:
jfyi; I've added a warning message about the requested memory being less than the default one.

Reply:
@mabartos this warning will happen on every reconciliation, so it will need to be at a debug level and/or a status warning instead.

Reply:
ok, thanks